### PR TITLE
docs(specs): add missing versioned consts

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/4010

### CODE REVIEW

File: specs/src/parameters_v3.md [hunk 1]
```diff
@@ -8,11 +8,15 @@ hardcoded in the application or they are blocked by the `x/paramfilter` module.
...
+| SquareSizeUpperBound | 128           | Hardcoded maximum square size which limits the number of shares per row or column for the original data square (not yet extended). | False                     |
+| SubtreeRootThreshold | 64            | See ADR 13 for more details.                                                                                                       | False                     |
+| MaxTxSize            | 2 MiB         | Maximum size of a transaction in bytes.                                                                                            | False                     |
+| TimeoutPropose       | 3500 ms       | Specifies the time that validators wait during the proposal phase of the consensus process. See CometBFT specs for more details.   | False                     |
+| TimeoutCommit        | 4200 ms       | Specifies the duration that validators wait during the Commit phase of the consensus process. See CometBFT specs for more details. | False                     |
```
Comment (rach-id) [inline specs/src/parameters_v3.md:17]: [nit]
maybe this is a good link to have
```suggestion
| TimeoutCommit        | 4200 ms       | Specifies the duration that validators wait during the Commit phase of the consensus process. See CometBFT [specs](https://github.com/celestiaorg/celestia-core/blob/v0.34.x-celestia/spec/consensus/consensus.md#precommit-step-heighthroundr) for more details. | False                     |
```

File: specs/src/parameters_v3.md [hunk 2]
```diff
@@ -8,11 +8,15 @@ hardcoded in the application or they are blocked by the `x/paramfilter` module.
...
+|----------------------|---------------|------------------------------------------------------------------------------------------------------------------------------------|---------------------------|
+| SquareSizeUpperBound | 128           | Hardcoded maximum square size which limits the number of shares per row or column for the original data square (not yet extended). | False                     |
+| SubtreeRootThreshold | 64            | See ADR 13 for more details.                                                                                                       | False                     |
+| MaxTxSize            | 2 MiB         | Maximum size of a transaction in bytes.                                                                                            | False                     |
+| TimeoutPropose       | 3500 ms       | Specifies the time that validators wait during the proposal phase of the consensus process. See CometBFT specs for more details.   | False                     |
```
Comment (rach-id) [inline specs/src/parameters_v3.md:16]: [nit]
```suggestion
| TimeoutPropose       | 3500 ms       | Specifies the time that validators wait during the proposal phase of the consensus process. See CometBFT [specs](https://github.com/celestiaorg/celestia-core/blob/v0.34.x-celestia/spec/consensus/consensus.md#propose-step-heighthroundr) for more details.   | False                     |
```

File: pkg/appconsts/v1/app_consts.go [hunk 3]
```diff
@@ -6,10 +6,16 @@ const (
 	Version              uint64 = 1
 	SquareSizeUpperBound int    = 128
 	SubtreeRootThreshold int    = 64
-	TimeoutPropose              = time.Second * 10
-	TimeoutCommit               = time.Second * 11
+	// Note: TimeoutPropose and TimeoutCommit weren't actually constants in v1,
+	// they were defaults for user-configurable timeouts. TODO: consider
```
Comment (rach-id) [inline pkg/appconsts/v1/app_consts.go:10]: apparently it's used in the v1 -> v2 tests:

```
func upgradeFromV1ToV2(t *testing.T, testApp *app.App) {
	t.Helper()
	testApp.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{
		Height:  2,
		Version: tmversion.Consensus{App: 1},
	}})
	endBlockResp := testApp.EndBlock(abci.RequestEndBlock{Height: 2})
	require.Equal(t, appconsts.GetTimeoutCommit(v1.Version),
		endBlockResp.Timeouts.TimeoutCommit)
	require.Equal(t, appconsts.GetTimeoutPropose(v1.Version),
		endBlockResp.Timeouts.TimeoutPropose)
	testApp.Commit()
	require.EqualValues(t, 2, testApp.AppVersion())
}
```

so it's fine to leave, I guess

Comment (rootulp) [reply pkg/appconsts/v1/app_consts.go:10]: Thanks! I'll remove the TODO.

**Review by rach-id — APPROVED** [review]

File: pkg/appconsts/v1/app_consts.go [hunk 4]
```diff
@@ -6,10 +6,17 @@ const (
...
+	// user-configurable timeout.
+	TimeoutCommit = time.Second * 11
 	// UpgradeHeightDelay is the number of blocks after a quorum has been
-	// reached that the chain should upgrade to the new version. Assuming a block
-	// interval of 12 seconds, this is 7 days.
+	// reached that the chain should upgrade to the new version. Assuming a
+	// block interval of 12 seconds, this is 7 days.
+	//
+	// TODO: why does this constant exist in v1? v1 does not contain the signal
+	// module.
```
Comment (cmwaters) [inline pkg/appconsts/v1/app_consts.go:20]: Yeah I think it should be removed

**Review by cmwaters — COMMENTED** [review]
On a slight tangent, did we end up adding the hardcoded heights for each of the networks for the v2 upgrade?

**Review by evan-forbes — APPROVED** [review]

**Review by ninabarbakadze — APPROVED** [review]

**Pull request merged**